### ISSUE TITLE
early stopping

### ISSUE THREAD

**Issue by leke-lyu** [mlandis/phyddle#9]
Hi Dr. Landis,

While reviewing the training code, I noticed that the early stopping criterion is based on epoch-to-epoch validation loss comparisons, rather than comparing the current validation loss to the global best validation loss observed so far.

I was wondering whether this design choice was intentional. For example, to encourage smoother convergence or faster stopping?

Best,
Leke Lyu

**Comment by mlandis** [comment]
Hi Leke Lyu,

Thanks -- great idea! I just pushed an update to the `development` branch (https://github.com/mlandis/phyddle/commit/4c317746ce60aa1be3d6d3750b659e2341438c59). Through that commit, training now logs the best validation score, and users control what rule is used to stop training through the `early_stopping_rule` option:

- `total` will stop if the current validation loss score beat the best validation score for a total of `num_early_stop` epochs
- `consecutive` will stop if current validation loss score worsens for `num_early_stop` epochs **in a row**

I also improved Train output to be more informative. I made `early_stopping_rule = total` and `num_early_stop = 10` the new defaults. I'll experiment with the setup some more, with the goal of adding it into the next version update.

Best,
Michael
